app/src/main/java/com/amos/flyinn/MainActivity.java

```diff
@@ -175,17 +178,41 @@ public void onClick(View view) {
         this.initViewsWebRTC();
         this.initScreenCapturePermissions();
     }
-
+ 
+ 
+ 
     /**
      * A native method that is implemented by the 'native-lib' native library,
      * which is packaged with this application.
      */
     public native String stringFromJNI();
-
+ 
     public void toScreenActivityOnClick(View view) {
         Intent intent = new Intent(this, RecordingActivity.class);
         startActivity(intent);
     }
+ 
+  
+ 
+    @Override
+    public void onResume()
+    {
+        super.onResume();
+        checkForDebuggingMode();
+    }
+ 
+     private void checkForDebuggingMode() {
+        if(Settings.Secure.getInt(this.getContentResolver(), Settings.Global.ADB_ENABLED, 0) != 1) {
+            try {
+                startActivity(new Intent(android.provider.Settings.ACTION_APPLICATION_DEVELOPMENT_SETTINGS));
```

Comment:
I, as a user, would prefer to be notified with an [AlertDialog](https://developer.android.com/guide/topics/ui/dialogs) about what is going to happen and what I am expected to do before the settings app is started.